web/src/compose_reply.js

```diff
@@ -184,8 +191,191 @@ export function quote_and_reply(opts) {
+    reference_div.style.height = `${relative_rect.height}px`;
+    // Allow clicks to go through.
+    reference_div.style.pointerEvents = "none";
+    message_content_div.append(reference_div);
+    return reference_div;
+}
+
+function show_quote_and_reply_popover(reference_element, message_id, selected_message_content_raw) {
+    popover_menus.toggle_popover_menu(reference_element, {
+        popperOptions: {
+            modifiers: [
+                {
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: ["bottom", "left", "right"],
+                    },
+                },
+            ],
+        },
+        // Add a delay to separate `hideOnClick` and `onShow` so that
+        // `onShow` is called after `hideOnClick`.
+        // See https://github.com/atomiks/tippyjs/issues/230 for more details.
+        delay: [100, 0],
+        onShow(instance) {
+            popover_menus.popover_instances.quote = instance;
+            instance.setContent(parse_html(render_quote_and_reply_popover()));
+            return true;
+        },
+        onHidden(instance) {
+            instance.destroy();
+            reference_element.remove();
+            popover_menus.popover_instances.quote = undefined;
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            $popper.on("click", ".quote_and_reply", (e) => {
+                quote_and_reply({
+                    message_id,
+                    selected_message_content_raw,
+                });
+                instance.hide();
+                e.preventDefault();
+                e.stopPropagation();
+            });
+        },
+    });
+}
+
+function suggest_quoting_message_selection(selection = window.getSelection()) {
+    if (!selection?.toString()) {
+        return;
+    }
+    const {start_id, end_id} = copy_and_paste.analyze_selection(selection);
+    // If selection spans multiple messages, do nothing.
+    if (start_id !== end_id) {
+        return;
+    }
+    const {selected_message_content_raw, selection_bounds} = get_message_selection(selection);
+
+    if (selected_message_content_raw) {
+        const message_id = start_id;
+        const reference_element = create_popover_reference_in_message(
+            document.querySelector(
+                `.focused-message-list [data-message-id="${message_id}"] .message_content`,
+            ),
+            selection_bounds,
+        );
+        show_quote_and_reply_popover(reference_element, message_id, selected_message_content_raw);
+    }
+}
+
 export function initialize() {
     $("body").on("click", ".compose_reply_button", () => {
         respond_to_message({trigger: "reply button"});
     });
+
+    $("body").on("mouseup", ".messagebox", () => {
+        // This gives the browser time to update the value returned by window.getSelection(),
+        // especially needed for cases where a click clears the previous selection.
```

Comment:
I'd spell out a bit more. E.g. "This is necessary to avoid showing the popover for whatever was selected before the browser before our click."